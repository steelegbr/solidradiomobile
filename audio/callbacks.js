/**
 * Audio player callbacks.
 */

 module.exports = async function() {
     
 }