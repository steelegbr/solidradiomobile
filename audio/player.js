/**
 * Indicates the possible player states.
 */

export const PlayerState = {
    UNINITIALISED: 0,
    IDLE: 1,
    LOADING: 3,
    PLAYING: 4,
    PAUSED: 5,
    ERROR: 6
}