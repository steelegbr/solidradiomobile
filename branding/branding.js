/**
 * Provides branding functions.
 */

import { DefaultTheme, DarkTheme } from 'react-native-paper';

/**
 * Generates a station theme.
 * @param {bool} darkMode An indication if we are in dark mode.
 * @param {station} station The station this theme is for.
 */

export function generateStationTheme(darkMode, station) {

    let theme = generateTheme(darkMode);
    theme.colors.primary = station.primary_colour;
    theme.colors.accent = station.primary_colour;
    theme.mode = 'adaptive';
    return theme;

}

/**
 * Generates a theme with no station.
 * @param {bool} darkMode An indication if we are in dark mode.
 */

export function generateTheme(darkMode) {

    if (darkMode) {
        return {
            ...DarkTheme,
            roundness: 10,
            colors: {
                ...DarkTheme.colors
            }
        };
    } else {
        return {
            ...DefaultTheme,
            roundness: 10,
            colors: {
                ...DefaultTheme.colors
            }
        };
    }

}
