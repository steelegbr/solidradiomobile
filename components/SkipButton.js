import React, { Component } from 'react';
import { Button } from 'react-native-paper';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/MaterialIcons';

export const DIRECTION_FORWARDS = 0;
export const DIRECTION_BACKWARDS = 1;

class SkipButton extends Component {

    render() {

        const { icon, theme, bigMode, iconSize } = this.props;

        if (bigMode) {
            return(
                <Button theme={theme}>
                    <Icon name={icon} size={iconSize} />
                </Button>
            );
        } else {
            return(
                <Button theme={theme}>
                    <Icon name={icon} size={16} />
                </Button>
            );
        }

    }

}

function mapStateToProps(state, ownProps) {

    const bigMode = 'bigMode' in ownProps ? ownProps.bigMode : false;
    const iconSize = 'iconSize' in ownProps ? ownProps.iconSize : 0;
    const icon = ownProps.direction == DIRECTION_FORWARDS ? "skip-next" : "skip-previous"

    return {
        icon: icon,
        theme: state.theme,
        bigMode: bigMode,
        iconSize: iconSize
    };

}

export default connect(mapStateToProps)(SkipButton);