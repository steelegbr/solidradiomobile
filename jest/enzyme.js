/**
    Solid Radio Mobile App
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import 'react-native';
import 'jest-enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

/**
 * Set up Enzyme to mount to DOM, simulate events,
 * and inspect the DOM in tests.
 */
Enzyme.configure({ adapter: new Adapter() });

// JSDOM - required for Enzyme's mount

const { JSDOM } = require('jsdom');
const jsdom = new JSDOM('<!doctype html><html><body></body></html>');
const { window } = jsdom;
global.window = window;
global.document = window.document;