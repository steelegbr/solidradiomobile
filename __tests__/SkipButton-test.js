/**
 * Tests the skip button.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import SkipButton from '../components/SkipButton';
import { PlayerState } from '../audio/player';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { mount } from 'enzyme';
import { generateTheme } from '../branding/branding';

// Redux

const mockStore = configureStore([]);
let store;

// Tests

describe('mid-playlist', () => {

    beforeEach(() => {
        store = mockStore({
            theme: generateTheme(false),
            player: {
                playlist: [ {}, {}, {} ],
                currentItem: 1
            }
        });

    });

    it('renders-forward-small', () => {

        // Arrange

        const skipButton = renderer.create(
            <Provider store={store}>
                <SkipButton direction={0} iconSize={12} />
            </Provider>
        ).toJSON();

        // Act

        // Assert

        expect(skipButton).toMatchSnapshot();

    });
    
    it('renders-forward-large', () => {

        // Arrange

        const skipButton = renderer.create(
            <Provider store={store}>
                <SkipButton direction={0} bigMode={true} size={40} />
            </Provider>
        ).toJSON();

        // Act

        // Assert

        expect(skipButton).toMatchSnapshot();
    
    });
    
    it('renders-backward-small', () => {

        // Arrange

        const skipButton = renderer.create(
            <Provider store={store}>
                <SkipButton direction={1} />
            </Provider>
        ).toJSON();

        // Act

        // Assert

        expect(skipButton).toMatchSnapshot();
    
    });
    
    it('renders-backward-large', () => {

        // Arrange

        const skipButton = renderer.create(
            <Provider store={store}>
                <SkipButton direction={1} bigMode={true} iconSize={40} />
            </Provider>
        ).toJSON();

        // Act

        // Assert

        expect(skipButton).toMatchSnapshot();
    
    });

    it('renders-unexpected-direction', () => {

        // Arrange
    
        const skipButton = renderer.create(
            <Provider store={store}>
                <SkipButton direction={5} />
            </Provider>
        ).toJSON();
    
        // Act
    
        // Assert
    
        expect(skipButton).toMatchSnapshot();
    
    });

    /*

    it('skip-pressed-forward', () => {

        // Arrange
    
        const skipButton = mount(
            <Provider store={store}>
                <SkipButton direction={0} />
            </Provider>
        );
    
        // Act
    
        skipButton.find(Icon.Button).first().props().onPress();
    
        // Assert
    
        const expectedActions = [ { "type": "AUDIO_PLAYER_PLAYPAUSE", "source": "play_pause_button" } ];
        expect(store.getActions()).toStrictEqual(expectedActions);
    
    });

    it('skip-pressed-backward', () => {

        // Arrange
    
        const skipButton = renderer.create(
            <Provider store={store}>
                <SkipButton direction={1} />
            </Provider>
        ).toJSON();
    
        // Act
    
        skipButton.find(Icon.Button).first().props().onPress();
    
        // Assert
    
        const expectedActions = [ { "type": "AUDIO_PLAYER_PLAYPAUSE", "source": "play_pause_button" } ];
        expect(store.getActions()).toStrictEqual(expectedActions);
    
    });

    */

});


it('renders-forward-disable', () => {

    // Arrange

    store = mockStore({
        theme: generateTheme(false),
        player: {
            playlist: [ {}, {}, {} ],
            currentItem: 2
        }
    });

    const skipButton = renderer.create(
        <Provider store={store}>
            <SkipButton direction={0} />
        </Provider>
    ).toJSON();

    // Act

    // Assert

    expect(skipButton).toMatchSnapshot();

});

it('renders-backward-disable', () => {

    // Arrange

    store = mockStore({
        theme: generateTheme(false),
        player: {
            playlist: [ {}, {}, {} ],
            currentItem: 0
        }
    });

    const skipButton = renderer.create(
        <Provider store={store}>
            <SkipButton direction={1} />
        </Provider>
    ).toJSON();

    // Act

    // Assert

    expect(skipButton).toMatchSnapshot();

});