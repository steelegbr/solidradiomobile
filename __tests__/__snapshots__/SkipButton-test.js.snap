// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mid-playlist renders-backward-large 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 40,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`mid-playlist renders-backward-small 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 16,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`mid-playlist renders-forward-large 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 0,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`mid-playlist renders-forward-small 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 16,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`mid-playlist renders-unexpected-direction 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 16,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders-backward-disable 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 16,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders-forward-disable 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "overflow": "hidden",
      },
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.0)",
        "borderRadius": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "padding": 8,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.0)",
          "borderRadius": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#6200ee",
            "fontSize": 16,
          },
          Object {
            "marginRight": 10,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
